pagerduty/resource_pagerduty_extension.go

```diff
@@ -174,3 +188,35 @@ func flattenExtensionObjects(serviceList []*pagerduty.ServiceReference) interfac
 	}
 	return services
 }
+
+func expandExtensionConfig(v interface{}) interface{} {
```

Comment:
Would it be possible to just use the generic https://github.com/hashicorp/terraform/blob/master/helper/structure/expand_json.go#L5 instead?

Reply:
The generic `ExpandJsonFromString()` function provided by terraform unmarshals JSON-encoded data into `map[string]interface{}` instead of `interface{}`

Same for `flattenExtensionConfig()` that marshals a `map[string]interface{}` value into JSON-encoded data.

I believe using those would require to change the type of the terraform configuration field `config` from `TypeString` to `TypeMap` *and* specify a schema for this map.
That last part is what I'm trying to avoid here as this would require to enumerate all possible configuration options.

Reply:
Ah makes sense :+1: let's keep it as-is then.